proxy-libs/helioslambda/lambda.go

```diff
@@ -22,6 +31,72 @@ type eventBridgeEvent struct {
 	TraceHeader string            `json:"trace-header"`
 }
 
+type sqsMessageCarrier struct {
+	messageAttrs map[string]events.SQSMessageAttribute
+}
+
+func (c sqsMessageCarrier) Get(key string) string {
+	if c.messageAttrs == nil {
+		return ""
+	}
+
+	for attrKey, val := range c.messageAttrs {
+		if attrKey == key {
+			return *val.StringValue
+		}
+	}
+
+	return ""
+}
+
+func (c sqsMessageCarrier) Set(key, val string) {
+	dataType := "String"
+	c.messageAttrs[key] = events.SQSMessageAttribute{DataType: dataType, StringValue: &val}
+}
+
+func (c sqsMessageCarrier) Keys() []string {
+	result := []string{}
+	for key := range c.messageAttrs {
+		result = append(result, key)
+	}
+
+	return result
+}
+
+func extractQueueName(eventSourceArn string) string {
+	if eventSourceArn != "" {
+		queueNameIndex := strings.LastIndex(":", eventSourceArn)
+		if queueNameIndex != -1 {
+			return eventSourceArn[queueNameIndex+1:]
+		}
+	}
+	return eventSourceArn
+}
+
+func HandleRecord(ctx context.Context, record events.SQSMessage, handleRecordHelper func(ctx context.Context, message events.SQSMessage) (any, error)) (any, error) {
+	recordCtx := propagator.Extract(ctx, sqsMessageCarrier{record.MessageAttributes})
+	tp := otel.GetTracerProvider()
+	messageId := attribute.KeyValue{
+		Key:   semconv.MessageIDKey,
+		Value: attribute.StringValue(record.MessageId),
+	}
+	messagingSystem := attribute.KeyValue{
+		Key:   semconv.MessagingSystemKey,
+		Value: attribute.StringValue("aws.sqs"),
+	}
+	messagingPayload := attribute.KeyValue{
+		Key:   "faas.event",
+		Value: attribute.StringValue(record.Body),
+	}
+	spanName := fmt.Sprintf("process %s", extractQueueName(record.EventSourceARN))
+	if spanName == "" {
```

Comment:
Span name will never be empty here, you need to check if the queue name is empty